core/loader/atom.ts

```diff
@@ -36,6 +37,69 @@ function parsePosts(text: TextResponse): OriginPost[] {
     })
 }
 
+/**
+ * Returns next or previous pagination url from feed xml, if present.
+ * See "paged feeds" https://www.rfc-editor.org/rfc/rfc5005#section-3
+ */
+function getPaginationUrl(
+  xmlResponse: TextResponse,
+  rel: 'first' | 'last' | 'next' | 'previous'
+): string | undefined {
+  let document = xmlResponse.parseXml()
+  if (!document) return undefined
+  let nextPageLink = [...document.querySelectorAll('link')].find(
+    link => link.getAttribute('rel') === rel
+  )
+  return nextPageLink ? buildFullURL(nextPageLink, xmlResponse.url) : undefined
+}
+
+type PostsCursor =
+  | [OriginPost[], PostsListLoader | undefined]
+  | [undefined, PostsListLoader]
+
+/**
+ * If xml response is ready, returns a tuple of posts and possibly
+ * the loader of the next portion of posts, if xml contains a link to them.
+ * If xml response is not yet ready, returns the recursive loader of posts.
+ */
+function getPostsCursor(
+  task: DownloadTask,
+  feedUrl: string,
+  feedResponse: TextResponse | undefined
+): PostsCursor {
+  if (!feedResponse) {
+    return [
+      undefined,
+      async () => {
+        let response = await task.text(feedUrl)
+        let [posts, loader] = getPostsCursor(task, feedUrl, response)
+        return [posts, loader] as [OriginPost[], PostsListLoader | undefined]
```

Comment:
Why do we need this type cast? Maybe we can just add extra `if` like `if (post && loader)`?

Reply:
Yep let's avoid type cast.